### ISSUE TITLE
Items gone with keepinventory 2736

### ISSUE THREAD

**Issue by Whazaaaaa** [SpongePowered/SpongeForge#1864]
When keepInventory is ON, the items in the players inventory dissapear. This didnt happen in 2734.

**I am currently running...**
- SpongeForge version: spongeforge-1.12.2-2529-7.0.0-BETA-2736
- Forge version: forge-1.12.2-14.23.0.2544-universal
- Java version: 8
- Operating System:
 - [x] Linux (Ubuntu, Debian, Fedora, etc)
 - [ ] macOS
 - [x] Windows
- Plugins/Mods: 
no plugins no mods

**What is the expected result?**
The items to stay in the inventory with the gamerule keepInventory on true
**What is the current result?**
Player lose all their items.

Simply put spongeforge 2736 on a 1.12.2 server (in my case forge-1.12.2-14.23.0.2544-universal) turn keepInventory on true and you see that items will be gone after death.

**Comment by Whazaaaaa** [comment]
Okay im not sure whats going on with inventories. But more people are losing their inventory. And from what I hear its not always because they died (with keepInventory on) So I'll just revert to 2734 untill this is solved.

**Comment by phit** [comment]
Most likely caused by https://github.com/SpongePowered/SpongeCommon/commit/211ba4dd20cc2e4858c4fcc83f67f76900d44815

**Comment by ryantheleach** [comment]
Duplicate of : https://github.com/SpongePowered/SpongeForge/issues/1870

**Comment by phit** [comment]
Solved in https://github.com/SpongePowered/SpongeCommon/commit/27e245433a9939c6d1ea930c7049b1a3d5961176
